prism_compile.c

```diff
@@ -6410,6 +6432,9 @@ pm_compile_node(rb_iseq_t *iseq, const pm_node_t *node, LINK_ANCHOR *const ret,
 
         // We set the index_lookup_table on the scope node so we can
         // refer to the parameters correctly
+        if (scope_node->index_lookup_table) {
+            st_free_table(scope_node->index_lookup_table);
+        }
```

Comment:
```suggestion
        pm_scope_node_destroy(scope_node);
```

Reply:
I specifically did not use `pm_scope_node_destroy` here because we're not destroying the scope node, but rather only overwriting the `index_lookup_table`. If we freed other things in `pm_scope_node_destroy`, we don't want those to get freed here.

Reply:
Got it, that makes sense!